react-components/src/components/SceneContainer/SceneContainer.tsx

```diff
@@ -0,0 +1,240 @@
+/*!
+ * Copyright 2023 Cognite AS
+ */
+import { type ReactElement, useEffect, useRef, useState } from 'react';
+import { type AddModelOptions } from '@cognite/reveal';
+import { useReveal } from '../RevealContainer/RevealContext';
+import * as THREE from 'three';
+import { type Query } from '../../utilities/FdmSDK';
+import { useFdmSdk } from '../RevealContainer/SDKProvider';
+import { createGetSceneQuery } from './Queries';
+import {
+  type GroundPlaneResponse,
+  type GroundPlaneProperties,
+  type SceneResponse,
+  type SkyboxProperties,
+  type GroundPlaneEdgeResponse,
+  type Transformation3d,
+  type SceneModelsProperties
+} from './SceneFdmTypes';
+import { type AddResourceOptions, Reveal3DResources } from '../..';
+import type CogniteClient from '@cognite/sdk/dist/src/cogniteClient';
+
+export type CogniteSceneProps = {
+  sceneExternalId: string;
+  sceneSpaceId: string;
+  sdk: CogniteClient;
+  onResourcesAdded?: () => void;
+  onResourceLoadError?: (error: any) => void;
+};
+
+export function SceneContainer({
+  sceneExternalId,
+  sceneSpaceId,
+  sdk
+}: CogniteSceneProps): ReactElement {
+  const viewer = useReveal();
+  const fdmSdk = useFdmSdk();
+  const skyboxRef = useRef<THREE.Object3D<THREE.Object3DEventMap>>();
+  const groundPlaneRef = useRef<Array<THREE.Object3D<THREE.Object3DEventMap>>>([]);
+  const resourceOptions = useRef<AddResourceOptions[]>([]);
+  const [loaded, setLoaded] = useState(false);
+
+  useEffect(() => {
+    const loadScene = async (query: Query): Promise<void> => {
+      // FDM query for scene
+      const res = await fdmSdk.queryNodesAndEdges({
```

Comment:
This and all other network code should be put in a separate hook with `useQuery`.